src/envoy/core.clj

```diff
@@ -1,7 +1,7 @@
 (ns envoy.core
   (:require [cheshire.core :as json]
             [clojure.data :refer [diff]]
-            [clojure.core.async :refer [go-loop go <! >! >!! alt! chan]]
+            [clojure.core.async :refer [go-loop go <! >! >!! alt! chan close!]]
```

Comment:
let's create an "`envoy.watches`" (as per [docs](https://developer.hashicorp.com/consul/docs/dynamic-app-config/watches)) namespace and move all the core.async and watcher specific functionality there

a few benefits:

1. `envoy.core` can be used with [babashka](https://github.com/babashka/babashka). it can't be used now because of "`core.async`"
2. we can later add other features / ops for watcher when/if we need to
3. decoupling is always good :)